### PR TITLE
feat:  swap asset

### PR DESCRIPTION
**Pull request by coldsnap666**

### CODE REVIEW

File: apps/ui/src/contexts/index.tsx [hunk 1]
```diff
@@ -23,10 +44,96 @@ export const GliaswapProvider: React.FC = (props) => {
...
+  walletConnectStatus: ConnectStatus;
+  // when a wallet was connected, the user lock would be filled
+  currentUserLock?: Script;
+
+  currentCkbAddress: string;
+
+  currentEthAddress: string;
+}
+
+export function useGliaswap(): GliaswapState {
```
Comment (homu-chain) [inline apps/ui/src/contexts/index.tsx:80]: the current useGliaswap is locale in `hooks/useGliaswap`, the `contexts` is not recommend using in components

File: apps/ui/src/suite/api/ServerGliaswapAPI.ts [hunk 2]
```diff
@@ -66,21 +111,65 @@ export class ServerGliaswapAPI implements GliaswapAPI {
     return Promise.resolve([]);
   }
 
-  async getAssetsWithBalance(lock: Script, assets?: ChainSpec[]): Promise<GliaswapAssetWithBalance[]> {
+  async getAssetsWithBalance(
+    lock: CkbScript,
+    assets?: Asset[],
+    ethAddr?: string,
```
Comment (homu-chain) [inline apps/ui/src/suite/api/ServerGliaswapAPI.ts:117]: `lock.args`  is enough, it is equals eth address when we use the pw lock

File: apps/ui/src/suite/api/ServerGliaswapAPI.ts [hunk 3]
```diff
@@ -66,21 +111,65 @@ export class ServerGliaswapAPI implements GliaswapAPI {
...
+    web3?: Web3,
```
Comment (homu-chain) [inline apps/ui/src/suite/api/ServerGliaswapAPI.ts:118]: we should move the dependencies to implement, for example `new GliaswapServerAndWeb3API(web3,...)`

File: apps/ui/src/suite/api/ServerGliaswapAPI.ts [hunk 4]
```diff
@@ -66,21 +111,65 @@ export class ServerGliaswapAPI implements GliaswapAPI {
...
       const res = await this.axios.post('/get-asset-with-balance', { lock });
       return res.data;
     }
 
+    // @ts-ignore
     const nervosChainSpecs = assets.filter(isCkbChainSpec).map(getCkbChainSpec);
-    const res = await this.axios.post<
-      (CkbNativeAssetWithBalance | CkbSudtAssetWithBalance | ShadowFromEthWithBalance)[]
+    const ckbAssets = await this.axios.post<
+      (CkbNativeAssetWithBalance | CkbSudtAssetWithBalance | ShadowOfEthWithBalance)[]
```
Comment (homu-chain) [inline apps/ui/src/suite/api/ServerGliaswapAPI.ts:128]: `ShadowOfXXX` is renamed to `ShadowFromXXX`

**Review by zzkouniya — APPROVED** [review]

**Pull request merged**